pkg/network/node/networkpolicy.go

```diff
@@ -406,9 +406,11 @@ func (np *networkPolicyPlugin) selectPods(npns *npNamespace, lsel *metav1.LabelS
 		utilruntime.HandleError(fmt.Errorf("ValidateNetworkPolicy() failure! Invalid PodSelector: %v", err))
 		return ips
 	}
-	for _, pod := range np.pods {
-		if (npns.name == pod.Namespace) && sel.Matches(labels.Set(pod.Labels)) {
-			ips = append(ips, pod.Status.PodIP)
+
+	// Iterate over the key instead of the value for performance reasons
+	for k := range np.pods {
```

Comment:
hm... would it be better to make `np.pods` be a map of `*kapi.Pod` rather than `kapi.Pod` instead? As it is now you're dereferencing `np.pods[k]` three times which can't be great?

In 4.x we dropped `np.pods` since the informer already has its own cache (https://github.com/openshift/sdn/commit/4bac7c7a568937dfe4680bb8d8357b8093ec616d + https://github.com/openshift/sdn/commit/eaf68af70cade6ea53663a0e24843149e9c63907) but that also required rewriting the unit tests to use a fake kubeclient (https://github.com/openshift/sdn/commit/2e60296cc8a78033234129b8a6e84e56b7dfed3a) and I'm not sure how much divergence there's been since 3.11 so that might be too much to backport...

Reply:
I took the easy path because I really don't know this code

Let me try to check how difficult are the backports

Reply:
the patches doesn't apply directly ... it is not straight forward to backport all that stuff